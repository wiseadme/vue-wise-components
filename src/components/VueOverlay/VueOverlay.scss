.vue-overlay {
  @include flex(center, center);
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  opacity: 0;
  transition: opacity .3s;

  &--hidden {
    opacity: 0;
  }

  &--active {
    opacity: .4;
    z-index: 100;
    transition: opacity .3s;
  }
}


