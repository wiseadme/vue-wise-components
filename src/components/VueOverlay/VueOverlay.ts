// Styles
import './VueOverlay.scss'

// Constructor
import Vue from 'vue'
import { VNode } from 'vue/types'

export default Vue.extend({
  name: 'VueOverlay',
  props: {
    hide: Boolean,
    active: Boolean,
    color:String,
    opacity: [Number, String]
  },

  computed: {
    classes(): Record<string, boolean> {
      return {
        ['vue-overlay--hidden']: this.hide,
        ['vue-overlay--active']: this.active,
        [this.color]: !!this.color
      }
    }
  },

  render(h): VNode {
    return h('div', {
      staticClass: 'vue-overlay',
      class: {
        ...this.classes
      },
      style: {opacity: this.opacity ? this.opacity : ''}
    }, this.$slots.default)
  }
})
