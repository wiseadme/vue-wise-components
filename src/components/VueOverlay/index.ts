import VueOverlay from './VueOverlay'

export default VueOverlay
