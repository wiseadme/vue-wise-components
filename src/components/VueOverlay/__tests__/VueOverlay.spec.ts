import { shallowMount } from '@vue/test-utils'
import VueOverlay from '../VueOverlay'

describe('VueOverlay.ts', () => {
  const color = 'teal--darken-1'
  const opacity = .9
  let wrapper: any = null

  beforeEach(() => {
    wrapper = shallowMount(VueOverlay, {
      propsData: {
        color,
        opacity
      }
    })
  })

  it('change overlay color by props value', () => {
    expect(wrapper.find(`.${ color }`).element).toBeInstanceOf(HTMLElement)
  })

  it('contains the color class', () => {
    expect(wrapper.find(`.${ color }`).element).toMatchSnapshot()
  })
})
