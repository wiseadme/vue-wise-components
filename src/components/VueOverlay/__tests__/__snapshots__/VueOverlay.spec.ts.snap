// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueOverlay.ts contains the color class 1`] = `
<div
  class="vue-overlay teal--darken-1"
  style="opacity: 0.9;"
/>
`;
